gslib/commands/signurl.py

```diff
@@ -112,12 +112,15 @@
   directly. This avoids the need to download the private key file.
 
 <B>OPTIONS</B>
-  -m           Specifies the HTTP method to be authorized for use
-               with the signed url, default is GET. You may also specify
-               RESUMABLE to create a signed resumable upload start URL. When
-               using a signed URL to start a resumable upload session, you will
-               need to specify the 'x-goog-resumable:start' header in the
-               request or else signature validation will fail.
+  -b <project> Allows you to specify a user project that will be billed for
+               requests that use signed URL. This is useful for generating
+               presigned links for buckets that use requester pays.
+
+               Note that it's not valid to specify both the ``-b`` and
+               ``--use-service-account`` options together.
```

Comment:
I think this still confuses me: why are -b and --use-service-account mutually exclusive (perhaps I don't understand correctly what --use-service-account does)?

Reply:
`--use-service-account` tells `signurl` to use a different API, so that rather than having to supply a keyfile, the command just uses the service account key to sign the URL. The phrasing of this could probably better represent this. I could also make it so that the option is just ignored rather than throwing an error. Something like "Note that the ``-b`` option will be ignored when generating keys with the ``--use-service-acount`` option." I could then print that message again when both options are used together, again rather than throwing an error.

Reply:
I like the idea of specifying that -b will be ignored if used along with -u.

Reply:
Wouldn't a problem with ignoring -b be that you could create signed URLs that seem valid, but then get rejected because the signed URL tries to access a Requester Pays bucket?